src/libraries/System.Text.Json/src/System/Text/Json/Serialization/PooledByteBufferWriter.cs

```diff
@@ -132,13 +132,23 @@ private void CheckAndResizeBuffer(int sizeHint)
                 sizeHint = MinimumBufferSize;
             }
 
-            int availableSpace = _rentedBuffer.Length - _index;
+            int currentLength = _rentedBuffer.Length;
+            int availableSpace = currentLength - _index;
```

Comment:
nit: Leave this as is and move the `int currentLength` local into the `if (sizeHint > availableSpace)` check.

```suggestion
            int availableSpace = _rentedBuffer.Length - _index;
```

[sharplab.io](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA0AXEBDAzgWwB8ABAJgEYBYAKGIAYACY8gOgCEBXAMy5ilwDcNGsQDMTUgwDCDAN40GihgqUAHKAEsAbtgwwGwAJ56A2gF0GAfVgA7PQBNOPPkOpKG67bv0a7V3/YwCK7uKoqeOnpMKNIAFjBgANYAgjb2AEowuBoAXjBOvFAAFL4YDNl5ABKlAJRhcvXuGlwMRRUw1X4AvF0MdHVu7kryg0ND7Z1lvaQArEghYwwAvsKjQ6UMYBxQthgAMjA2AOYYsQy91ocOBXwsB8enC2Mb2DoaADbYwO8wAMqq2DA+l6Wx2V3uJzOcH8aSCrkaSmarQmGwAfAxXtgPl8fv9ATABosGmsiRsjlAIAB3NiGc4MACyuliLEZCDauQ6pTQm22uwhpxq8JJiw2NhglN+HLpoL5h0hDAA1AxyVSaUKiU0WkUihxamKJVL0aUWdgEAA1bDvDgEhFjEYaxb6yV5aW88Fy06K8ocyZPB2IrU6vXi536I12E3my3Wwn+9z2uOOkNS3rG1kWq0wP1xlbCsa521DIymCwQd6ObiFOmXOwwCvOKDq/0166Vvh05I7bCGAAKEDLAB5izBUSxfrFsLB7CxMnYik6OYLVnG8TYh8YRx5YFoNBAOLgblA6WX64UWMlcKuinRuZYAkEl3mhuoYDu9we21AWFIIKpDAAVCAihbOtD0fOMXzffdD2/H5JyKcD/U7KBuz7Qdh1HcdJzrGcYAwbYbCKE8wOzJRc3ccilHqCJvGiOIEhSNJMnaQ9LFwDgjiOLIHBKPwUTsWNFATZ4tX4qZen6QthI1MS6VmeZCwLJ8XjeT5vj+AEgWrXZQM/O4PShGFAmCZcRVEn00QxVScQ0/FBKGaTST8GV3QeM4Lh009bn5WJSOePwVWpWlekZU5I3ZKouR5ME7B8x9CyaPwFxdEE3VigyvUCtVTIdJFtV1ATkrDBg01NDMYwS4ZKrGIrXRi/YMqVMS/I1PKg0K5MXXDDBI3Km0nyJRzE0UWrUwjdNoyzaqhkoh0lLjYdzAYYjP20q5dIbJsHRAryj16ZDUP7d51z0TCJynXC5yK+KBqUVcTs3SDd2g1behWhtz0vAFCJvIyHxa9wnvfGCfz/QDgM8kjpqBl6PqkODikQh0Dt7I6HrO7Dp0yfCoEI97CiRmb6lzJYgA=)